drivers/docker/driver_test.go

```diff
@@ -2911,28 +2911,159 @@ func TestDockerDriver_memoryLimits(t *testing.T) {
+	}
+	require.NoError(t, stopTask.EncodeConcreteDriverConfig(&stopSigTaskCfg))
+
+	stopKillTask := &drivers.TaskConfig{
+		ID:        uuid.Generate(),
+		Name:      "stopsig-killsig",
+		AllocID:   uuid.Generate(),
+		Resources: basicResources,
+	}
+	require.NoError(t, stopKillTask.EncodeConcreteDriverConfig(&stopSigTaskCfg))
+
+	killTask := &drivers.TaskConfig{
+		ID:        uuid.Generate(),
+		Name:      "killsig-only",
+		AllocID:   uuid.Generate(),
+		Resources: basicResources,
+	}
+	require.NoError(t, killTask.EncodeConcreteDriverConfig(&defaultTaskCfg))
 
-	t.Run("windows conversion", func(t *testing.T) {
-		s, err := d.parseSignal("windows", "SIGINT")
+	defaultTask := &drivers.TaskConfig{
+		ID:        uuid.Generate(),
+		Name:      "nosigs-default",
+		AllocID:   uuid.Generate(),
+		Resources: basicResources,
+	}
+	require.NoError(t, defaultTask.EncodeConcreteDriverConfig(&defaultTaskCfg))
+
+	type taskInfo struct {
+		killSig      string
+		expectedSigs []string
+	}
+
+	taskList := map[*drivers.TaskConfig]taskInfo{
+		stopTask:     {killSig: "", expectedSigs: []string{"19", "9"}},
+		defaultTask:  {killSig: "", expectedSigs: []string{"15", "9"}},
+		stopKillTask: {killSig: "SIGTERM", expectedSigs: []string{"15", "19", "9"}},
+		killTask:     {killSig: "SIGTERM", expectedSigs: []string{"15", "15", "9"}},
+	}
+
+	d := dockerDriverHarness(t, nil)
+
+	for task := range taskList {
+		cleanup := d.MkAllocDir(task, true)
+		defer cleanup()
+
+		if task.Name == "stopsig-only" || task.Name == "stopsig-killsig" {
+			copyImage(t, task.TaskDir(), "busybox_stopsignal.tar") // Default busybox image with STOPSIGNAL 19 added
+		} else {
+			copyImage(t, task.TaskDir(), "busybox.tar")
+		}
+	}
+
+	sigList := map[string][]string{} // maps Name-AllocID to signal slices
+
+	client := newTestDockerClient(t)
+	var err error
+
+	listener := make(chan *docker.APIEvents)
+	err = client.AddEventListener(listener)
+	require.NoError(t, err)
+
+	defer func() {
+		err = client.RemoveEventListener(listener)
 		require.NoError(t, err)
-		require.Equal(t, syscall.SIGTERM, s)
-	})
+	}()
 
-	t.Run("not a signal", func(t *testing.T) {
-		_, err := d.parseSignal(runtime.GOOS, "SIGDOESNOTEXIST")
-		require.Error(t, err)
-	})
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	go func() {
+		for {
+			select {
+			case msg := <-listener:
+				// Only add kill signals
+				if msg.Action == "kill" {
+					name := msg.Actor.Attributes["name"]
+					sig := msg.Actor.Attributes["signal"]
+					sigList[name] = append(sigList[name], sig)
+				}
+			case <-ctx.Done():
+				return
+			}
+		}
+	}()
+
+	for task, info := range taskList {
+		t.Run(task.Name, func(*testing.T) {
+			_, _, err = d.StartTask(task)
+			require.NoError(t, err)
+			require.NoError(t, d.WaitUntilStarted(task.ID, 5*time.Second))
+			err = d.StopTask(task.ID, 1*time.Second, info.killSig)
+
+			actualSigs := sigList[fmt.Sprintf("%s-%s", task.Name, task.AllocID)]
```

Comment:
`sigList` is accessed in multiple goroutines without locking. This might cause the test to panic depending on how it's run.  Let's pair on how to improve it.